### PR TITLE
Add TypeScript Type Definitions for @stellar-asset-lists/sdk

### PR DESCRIPTION
**Pull request by yripper**
## Description
This pull request introduces TypeScript type definitions for `@stellar-asset-lists/sdk`, enhancing its usability in TypeScript projects. By providing these types, developers can benefit from type checking, code completion, and documentation directly in their IDE, leading to a more intuitive and error-free development experience.

### Key Changes
- Added `index.d.ts`: Contains type definitions for the main exported functions (`fetchAvailableAssetLists` and `fetchAssetList`) along with interfaces for `AssetListDescriptor`, `AssetList`, and `Asset`.
- Updated `package.json`: Included the `types` field to point to the new type definitions file, making it discoverable by TypeScript.

### Motivation
In integrating `@stellar-asset-lists/sdk` into [stellartools.xyz](https://stellartools.xyz), the absence of type definitions highlighted a crucial gap. Contributing these types aims to enhance development efficiency, reduce potential errors, and improve the overall TypeScript ecosystem compatibility, benefiting all developers using the library in TypeScript environments.

### Testing
The added type definitions have been tested in a local TypeScript project to ensure they correctly represent the library's data structures and function signatures. Here are the steps followed for testing:
1. Linked the modified `@stellar-asset-lists/sdk` package locally using `yarn link`.
2. Integrated the linked package into a TypeScript project and ensured the compiler could successfully find and use the new type definitions without any errors.
3. Verified that the types are accurate and match the library's current functionality by calling the exported functions and constructing objects based on the defined interfaces.

### Request for Feedback
I welcome any feedback on the added type definitions, particularly regarding:
- Any discrepancies between the type definitions and the library's actual behavior.
- Suggestions for improving type coverage or accuracy.
- General feedback on the structure and clarity of the newly added types.

### CODE REVIEW

**Review by silence48 — APPROVED** [review]
Looks good to me with the update.

**Review by orbitlens — APPROVED** [review]

**Comment by orbitlens** [comment]
Thanks for adding types bindings!

**Comment by nimrodotsile** [comment]
> Thanks for adding types bindings!

ZxDnz5Ghig84NT9PNd7gixP2YvKqKbjPYVq6J5x2fMdng7NeEqWZNueUXJDqH9yFK2Gte7s1NxUAUcNuAgMGkauj2beXrXFXK

**Comment by nimrodotsile** [comment]
> > Thanks for adding types bindings!
> 
> ZxDnz5Ghig84NT9PNd7gixP2YvKqKbjPYVq6J5x2fMdng7NeEqWZNueUXJDqH9yFK2Gte7s1NxUAUcNuAgMGkauj2beXrXFXK

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+export type AssetListDescriptor = {
```
Comment (nimrodotsile) [inline index.d.ts]: ZxDnz5Ghig84NT9PNd7gixP2YvKqKbjPYVq6J5x2fMdng7NeEqWZNueUXJDqH9yFK2Gte7s1NxUAUcNuAgMGkauj2beXrXFXK

**Review by nimrodotsile — APPROVED** [review]
ZxDnz5Ghig84NT9PNd7gixP2YvKqKbjPYVq6J5x2fMdng7NeEqWZNueUXJDqH9yFK2Gte7s1NxUAUcNuAgMGkauj2beXrXFXK

**Pull request merged**